### PR TITLE
fix(sdk/sync): don't panic on error if we're canceling anyways

### PR DESCRIPTION
**Pull request by Stebalien**
related to #547

### CODE REVIEW

**Comment by Stebalien** [comment]
What other panics?

**Comment by aschmahmann** [comment]
#547 lists 3 types of errors:
1) Context cancellation (covered by this PR)
2) Redis lookup (DNS?) event
3) I/O timeouts

Issues 2 + 3 are failures happening in the redis client that probably shouldn't happen which is why we're panicking. However, since we are currently panicking the tests need some protection in the meanwhile (e.g. retries).

**Comment by Stebalien** [comment]
The second issue could easily be context cancellation as well. Do you have exact error messages for all of these?

File: sdk/sync/write.go [hunk 1]
```diff
@@ -76,25 +76,31 @@ func (w *Writer) keepAliveWorker(ctx context.Context) {
 	for {
 		select {
 		case <-time.After(KeepAlivePeriod):
-			w.keepAlive(ctx)
+			if err := w.keepAlive(ctx); err != nil {
+				if ctx.Err() != nil {
```
Comment (raulk) [inline sdk/sync/write.go]: Could possibly use:

```go
if errors.Is(err, context.Canceled)
```

**Comment by aschmahmann** [comment]
Only the second one, the third one is rarer and I forgot to record the error. What type of error message for the second one are you looking for? Is it more than what's reported in the other issue (copied below).

{"ts":1581618676264268161,"msg":"","group_id":"cypress","run_id":"5e5397620cd1","event":{"type":"finish","outcome":"failed","error":"dial tcp: lookup redis-headless on 100.64.0.10:53: write udp 100.96.12.27:59536->100.64.0.10:53: write: invalid argument"}}

Is it normal/reasonable that the send port will change in between requests? i.e. 100.96.12.27:53060 for one request and 100.96.12.27:59536 for another.

**Comment by Stebalien** [comment]
So, the errors are:

* Context canceled
* "dial tcp: lookup redis-headless on 100.64.0.10:53: write udp 100.96.12.27:59536->100.64.0.10:53: write: invalid argument"
* Some IO timeout error.

**Comment by Stebalien** [comment]
> Is it normal/reasonable that the send port will change in between requests? i.e. 100.96.12.27:53060 for one request and 100.96.12.27:59536 for another.

For DNS, yes.

File: sdk/sync/write.go (hunk 1 above)
Comment (Stebalien) [reply sdk/sync/write.go]: So... I don't trust errors. I'm seeing an "invalid argument" error which could, maybe, be caused by:

```go
go func() {
  <-ctx.Done()
  myConn.Close()
}()
return myConn.Read()
```

(depending on the connection type, etc.)

**Pull request merged**